common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -508,6 +508,10 @@ def import_xblock(self, user_id, course_key, block_type, block_id, fields=None,
                 draft_course = course_key.for_branch(ModuleStoreEnum.BranchName.draft)
                 with self.branch_setting(ModuleStoreEnum.Branch.draft_preferred, draft_course):
                     draft_block = self.import_xblock(user_id, draft_course, block_type, block_id, fields, runtime)
+                    # if block was published once and now it is in draft state then return draft version
+                    # as current state of block is draft state
+                    if self.has_published_version(draft_block) and block_type not in DIRECT_ONLY_CATEGORIES:
```

Comment:
Small nit: It'd be nice to pull this `block_type in DIRECT_ONLY_CATEGORIES` check into a descriptive method, like `is_auto_published_type(block_type)` - it reads better.